articles/active-directory/develop/msal-net-aad-b2c-considerations.md

```diff
@@ -68,14 +70,16 @@ Acquiring a token for an Azure AD B2C protected API in a public client applicati
 
 ```csharp
 IEnumerable<IAccount> accounts = await application.GetAccountsAsync();
-AuthenticationResult ar = await application .AcquireToken(scopes, parentWindow)
+AuthenticationResult ar = await application .AcquireTokenInteractive(scopes)
                                             .WithAccount(GetAccountByPolicy(accounts, policy))
+                                            .WithParentActivityOrWindow(ParentActivityOrWindow)
                                             .ExecuteAsync();
 ```
 
 with:
 
 - `policy` being one of the previous strings (for instance `PolicySignUpSignIn`).
+- `ParentActivityOrWindow` is an optional parameter, for platforms supporting it, the parent UI (window in Windows, Activity in Android, UIViewController in iOS). For Android (all flows) and iOS (broker only) this parameter is required. See more information [here on the UI dialog](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Acquiring-tokens-interactively#withparentactivityorwindow).
```

Comment:
This would seem to indicate that for Android, it's optional (emphasis mine):

> `ParentActivityOrWindow` is an **optional parameter**, for platforms supporting it, the parent UI (window in Windows, **Activity in Android**, UIViewController in iOS).

Yet, in the next sentence, it's  required:

> For Android (all flows) and iOS (broker only) this parameter is required.

I'm not familiar with the programming model on this platform, but perhaps can be reworded to be a bit clearer?

Reply:
I see...thanks for the feedback @marsh Macy good point. Android requires an activity, as does windows, but it's optional in iOS, but now w/our broker support in iOS, developers need to set the window for iOS as well. So it's a little confusing. Will fix the wording. 

---
In reply to: [340109685](https://github.com/MicrosoftDocs/azure-docs/pull/41631#discussion_r340109685) [](ancestors = 340109685)